### ISSUE TITLE
Toggle field and CSS issue

### ISSUE THREAD

**Issue by BeholdPL** [themeum/kirki#2146]
I have problem with toggle field with CSS output like this:

```
Kirki::add_field('behold-basic', [
    'type' => 'toggle',
    'settings' => 'show_header_top_bar_on_scroll',
    'label' => esc_html__('Pokaż pasek górny przy przewijaniu', 'behold-basic'),
    'section' => 'header_top_bar',
    'default' => '1',
    'priority' => 10,
    'transport' => 'auto',
    'output' => [
        [
            'element'       => '.on_scroll .l-header-nav-contacts',
            'property'      => 'max-height',
            'value_pattern' => '0',
            'exclude'       => [ true, 1, '1', 'true' ]
        ],
        [
            'element'       => '.on_scroll .l-header-nav-contacts',
            'property'      => 'border',
            'value_pattern' => 'none',
            'exclude'       => [ true, 1, '1', 'true' ]
        ],
        [
            'element'       => '.on_scroll .l-header-nav-contacts',
            'property'      => 'max-height',
            'value_pattern' => '999px',
            'exclude'       => [ false, 0, '0', 'false' ]
        ],
        [
            'element'       => '.on_scroll .l-header-nav-contacts',
            'property'      => 'border',
            'value_pattern' => '1px solid',
            'exclude'       => [ false, 0, '0', 'false' ]
        ],
    ],
]);
```

the problem is 'max-height' attribute. When toggle is on it should be "999px" and it is, but when Is off, should be "0" and this is working only on preview. When I reload site only "border: none" is added. Max-height isn't overwriting by other CSS either. What could be wrong?

**Comment by aristath** [comment]
I haven't tested this but looking at your code I'd try using `'0px'` instead of `'0'` for the max-height value.
I vaguely remember seeing something like that months ago, I think that should do the trick.
There's a check somewhere that checks if the value is empty and only prints CSS for settings that have a value. PHP sometimes considers `0` as an empty value depending on how you do the actual checking, so that may be the cause. If you use `0px` that is a non-numeric string so it's not considered "empty".
If this fixes your issue let me know and I'll mark this ticket as a bug so I can fix the `0` value comparison in the next update.

**Comment by DannyCooper** [comment]
Confirmed the 0px fix does work.

I think the issue could be fixed by replacing:

https://github.com/kirki-framework/module-css/blob/96b59f65100233a59ac049341c4e871a2e1600fe/src/CSS/Output.php#L108

with:

`if ( isset( $output['value_pattern'] ) && strlen( $output['value_pattern'] ) && is_string( $output['value_pattern'] ) ) {`

**Comment by aristath** [comment]
I checked this one and the `empty` check really doesn't do anything...
If no `value_pattern` is defined, then it falls-back to `$` which is not an empty string.
If a user adds an empty string then we have to assume it's by choice.

I removed the check entirely (didn 't replace it with `strlen`, this should be OK now.
Fix was pushed to develop branch, will be included in v4.0
